tests/drivers/flash/stm32/src/main.c

```diff
@@ -278,6 +278,12 @@ ZTEST(flash_stm32, test_stm32_block_registers)
 	TC_PRINT("Try to unlock blocked OPT\n");
 	__set_FAULTMASK(1);
 	flash_stm32_option_bytes_lock(flash_dev, false);
+
+	/* Ensure instruction pipeline is flushed to prevent imprecise
+	 * Bus Fault on STM32H7 with cache enabled
+	 */
```

Comment:
If my understanding is correct:
```suggestion
	/* Ensure the Imprecise Bus Fault caused by the illegal
	 * access is seen now while BusFault is still masked by
	 * triggering a Context synchronization event. This is
	 * notably required on series such as STM32H7 when Icache
	 * is enabled - the Imprecise Bus Fault is reported after
	 * we unmask BusFault and triggers a kernel panic.
	 */
```